### PR TITLE
opentelemetry: set status code

### PR DESCRIPTION
**Pull request by zirain**
Commit Message: set span status code base on http status code for OpenTelemetry tracer, xref: https://github.com/istio/istio/issues/50195
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue] #32114
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: source/extensions/tracers/opentelemetry/tracer.cc
Comment (silverstar194) [inline source/extensions/tracers/opentelemetry/tracer.cc:127]: Can you check the formatting here? The comment looks like it has strange line breaks.

Comment (zirain) [reply source/extensions/tracers/opentelemetry/tracer.cc:127]: reformated by `tools/local_fix_format.sh`

File: test/extensions/tracers/opentelemetry/opentelemetry_tracer_impl_test.cc
Comment (adisuissa) [inline test/extensions/tracers/opentelemetry/opentelemetry_tracer_impl_test.cc:567]: Consider adding this in its own explicit test (avoiding a test covering multiple use-cases).

File: source/extensions/tracers/opentelemetry/tracer.cc
Comment (adisuissa) [inline source/extensions/tracers/opentelemetry/tracer.cc:141]: Just to clarify, is it intentional to have both the attribute and the status code?
I see the benefit when introducing a new feature, just want to better understand whether this is desired.

**Review by adisuissa — COMMENTED** [review]
Overall LGTM.
Please add a release note.
Assigning @htuch as codeowner.
/assign @htuch

File: source/extensions/tracers/opentelemetry/tracer.cc
Comment (zirain) [reply source/extensions/tracers/opentelemetry/tracer.cc:141]: I confirmed with a lot of imeplements in OTel-collector, most of them set both.

File: changelogs/current.yaml
Comment (htuch) [inline changelogs/current.yaml:15]:
```suggestion
    Set status code for OpenTelemetry tracers (previously unset).
```

**Review by htuch — DISMISSED** [review]
LGTM modulo nit comment.

**Review by wbpcode — APPROVED** [review]

**Pull request merged**
